python/lsst/pipe/tasks/visualizeVisit.py

```diff
@@ -0,0 +1,250 @@
+    )
+
+    outputExp = cT.Output(
+        name="calexpBin",
+        doc="Output binned image.",
+        storageClass="ExposureF",
+        dimensions=("instrument", "detector"),
+    )
+
+
+class VisBinConfig(pipeBase.PipelineTaskConfig,
+                   pipelineConnections=VisBinConnections):
+    """Configuration for focal plane visualization.
+    """
+    binning = pexConfig.Field(
+        dtype=int,
+        default=8,
+        doc="Binning factor to apply.",
+    )
+    detectorKeyword = pexConfig.Field(
+        dtype=str,
+        default='DET-ID',
+        doc="Metadata keyword to use to find detector if not available from input.",
+    )
+
+
+class VisBinTask(pipeBase.PipelineTask,
+                 pipeBase.CmdLineTask):
+    """Task for focal plane visualization.
+    """
+    ConfigClass = VisBinConfig
+    _DefaultName = 'VisBin'
+
+    def run(self, inputExp, camera):
+        """Bin input image, attach associated detector.
+
+        Parameters
+        ----------
+        inputExp : `lsst.afw.image.Exposure`
+            Input exposure data to bin.
+        camera : `lsst.afw.cameraGeom.Camera`
+            Input camera to use for mosaic geometry.
+
+        Returns
+        -------
+        output : `lsst.pipe.base.Struct`
+            Results struct with attribute:
+
+            ``outputExp``
+                Binned version of input image. (`lsst.afw.image.Exposure`)
+        """
+        if inputExp.getDetector() is None:
+            detectorId = inputExp.getMetadata().get(self.config.detectorKeyword)
+            if detectorId is not None:
+                inputExp.setDetector(camera[detectorId])
+
+        binned = inputExp.getMaskedImage()
+        binned = afwMath.binImage(binned, self.config.binning)
+        outputExp = afwImage.makeExposure(binned)
+
+        if outputExp.getDetector() is None:
+            outputExp.setDetector(inputExp.getDetector())
+        return pipeBase.Struct(
+            outputExp=outputExp,
+        )
+
+
+class VisMosaicConnections(pipeBase.PipelineTaskConnections,
+                           dimensions=("instrument", )):
+    inputExps = cT.Input(
+        name="calexpBin",
+        doc="Input binned images mosaic.",
+        storageClass="ExposureF",
+        dimensions=("instrument", "detector"),
+        multiple=True,
+    )
+    camera = cT.PrerequisiteInput(
+        name="camera",
+        doc="Input camera to use for mosaic geometry.",
+        storageClass="Camera",
+        dimensions=("instrument", "calibration_label"),
+    )
+
+    outputData = cT.Output(
+        name="calexpFP",
+        doc="Output binned mosaicked frame.",
+        storageClass="ImageF",
+        dimensions=("instrument", ),
+    )
+
+
+class VisMosaicConfig(pipeBase.PipelineTaskConfig,
+                      pipelineConnections=VisMosaicConnections):
+    """Configuration for focal plane visualization.
+    """
+    binning = pexConfig.Field(
```

Comment:
So this is binning done on the focal plane exposure that is done from binned exposures? So if you bin exposures by 8, and then set this binning to 8 the final is 64 times smaller? This seems a bit unclear. If I am correct (or if I am not) please write more doc string to describe this, and maybe put it in comments. Someone just reading the configs might guess that this value must be set to match the other from the other task for some reason.

Reply:
It is multiplicative.  I'll update the docstring to make that clear, and to mention why this is a good thing.

Reply:
I've updated the docstring again to fix my incorrect understanding, after some additional testing from Yusra and Lee.  The two numbers must agree between tasks, and so I've added a contract enforcing that to the default pipeline.